### ISSUE TITLE
Add support for `postcss-scss` in `declaration-nested-properties` rule

### ISSUE THREAD

**Issue by jhae-de** [stylelint-scss/stylelint-scss#1264]
The `declaration-nested-properties` rule does not report an error if you extend the `stylelint-config-standard-scss` configuration and enable it with the "never" option.

The following [documented](https://github.com/stylelint-scss/stylelint-scss/tree/v7.0.0/src/rules/declaration-nested-properties#never) examples should report an error, but they don't:

```scss
a {
  margin: {
    left: 10px;
  }
}
```

```scss
a {
  background: red {
    repeat: no-repeat;
  }
}
```

Similar to the issue described [here](https://github.com/stylelint-scss/stylelint-scss/issues/1262), `postcss-scss` support is missing. There is also a corresponding commented-out `reject` [test case](https://github.com/stylelint-scss/stylelint-scss/blob/v7.0.0/src/rules/declaration-nested-properties/__tests__/index.js#L361) that fails when enabled.

**Comment by kristerkari** [comment]
Good that you found these, they are quite easy to miss when the tests are not covering the `postcss-scss` support 👍
